plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/mypy/MypyAnalysis.java

```diff
@@ -203,30 +266,45 @@ public void afterRunProcess(String output, String errors, IExternalCodeAnalysisS
                     if (severityFound.equals("warning")) {
                         markerSeverity = IMarker.SEVERITY_WARNING;
                     }
+                    if (severityFound.equals("note")) {
+                        markerSeverity = IMarker.SEVERITY_INFO;
+                    }
                     if (markerSeverity != -1) {
-                        line = Integer.parseInt(outputLine.substring(m.start(2), m.end(2)));
+                        line = Integer.parseInt(outputLine.substring(m.start(2), m.end(2))) - 1;
                         column = Integer.parseInt(outputLine.substring(m.start(3), m.end(3))) - 1;
                         messageId = "mypy";
                         message = outputLine.substring(m.start(5), m.end(5)).trim();
+
+                        IRegion region = null;
+                        try {
+                            region = document.getLineInformation(line);
+                        } catch (Exception e) {
+                        }
+                        if (region != null && document != null) {
+                            LineCol lineCol = new LineCol(line, column);
+                            if (lineColToMessage.containsKey(lineCol)) {
```

Comment:
This is just a minor nitpick.

It's usually better to `get` what you want from the map and check if it's `null` or not and then act on the instance received instead of doing a `containsKey` and then doing a new call to `get` to actually get it as you're doing 2 checks instead of one.